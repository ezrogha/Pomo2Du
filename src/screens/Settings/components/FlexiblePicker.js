import React from 'react'
import { Picker } from 'react-native'

export default ({ selectedValue, updatePicker, itemArray, metric }) => {
    const renderPickerItems = () => {
        return itemArray.map((value, index) => {
            return <Picker.Item key={index} label={`${value} ${metric}`} value={value} />
        })
    }
    
    return (
        <Picker selectedValue={selectedValue} onValueChange={updatePicker} >
            {renderPickerItems()}
        </Picker>
    )
}

