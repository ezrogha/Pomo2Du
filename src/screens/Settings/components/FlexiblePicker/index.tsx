import React from 'react'
import { Picker } from 'react-native'

interface Props {
    itemArray: string[],
    selectedValue: number,
    metric: number | string,
    updatePicker: (value: string) => void
}

const FlexiblePicker: React.FC<Props> = ({ selectedValue, updatePicker, itemArray, metric }) => {
    const renderPickerItems = () => {
        return itemArray.map((value, index) => {
            return <Picker.Item key={index} label={`${value} ${value !== "Custom" ? metric : ''}`} value={value} />
        })
    }
    
    return (
        <Picker selectedValue={selectedValue} onValueChange={updatePicker} >
            {renderPickerItems()}
        </Picker>
    )
}

export default FlexiblePicker