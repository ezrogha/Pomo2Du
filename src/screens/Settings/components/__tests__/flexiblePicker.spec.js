import React from 'react'

import FlexiblePicker from '../FlexiblePicker';
import renderer from 'react-test-renderer'

describe("Testing FlexiblePicker Component", () => {
    test("renders correctly", () => {
        const tree = renderer.create(<FlexiblePicker selectedValue={0} updatePicker={() => {}} itemArray={[0,1,2]} metric={"minutes"} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
