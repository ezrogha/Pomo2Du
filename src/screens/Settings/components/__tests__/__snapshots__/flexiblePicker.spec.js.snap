// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FlexiblePicker Component renders correctly 1`] = `
<View>
  <RCTPicker
    items={
      Array [
        Object {
          "label": "0 minutes",
          "textColor": undefined,
          "value": 0,
        },
        Object {
          "label": "1 minutes",
          "textColor": undefined,
          "value": 1,
        },
        Object {
          "label": "2 minutes",
          "textColor": undefined,
          "value": 2,
        },
      ]
    }
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    selectedIndex={0}
    style={
      Array [
        Object {
          "height": 216,
        },
        undefined,
      ]
    }
  />
</View>
`;
